### ISSUE TITLE
iOS Device not found

### ISSUE THREAD

**Issue by llorenzo** [Gamua/Adobe-Runtime-Support#374]
## Problem Description
iOS Device not found when debugging on device in Intellij

* AIR SDK Version 33.1.1.176
Environment:MacOS Mojave, Intellij, iOS
* Can you reproduce it yourself, or did you get reports from other users? If so, how many users are affected (percentage)? 100%

## Steps to Reproduce
Debug on iOS Device in Intellij.

## Known Workarounds
Revert to AIR 33.1.1.98

**Comment by stoff99** [comment]
Same problem here with MacOS 10.15.5 and IntelliJ 1.2.
With AIR SDK 33.1.1.176 no devices are found anymore.

**Comment by stoff99** [comment]
When you run  ./adt -devices -platform ios in the terminal you will get following error:

`dyld: Library not loaded: @loader_path/libplist.3.dylib
  Referenced from: /Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/idb
  Reason: no suitable image found.  Did find:
	/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/libplist.3.dylib: code signature in (/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/libplist.3.dylib) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
	/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/libplist.3.dylib: stat() failed with errno=1
	/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/libplist.3.dylib: code signature in (/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/libplist.3.dylib) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
	/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/libplist.3.dylib: stat() failed with errno=1
	file system relative paths not allowed in hardened programs
/var/folders/l6/__l5nsmx2rg9yq05ty4wdcz80000gn/T/cmd3408969850125211040.tmp: line 1: 27600 Abort trap: 6           "/Users/stefan/Development/Frameworks/AS3/Air_33_1_1_176/lib/aot/bin/iOSBin/idb" "-devices"`


With the 86 version you get:
List of attached devices:
Handle  DeviceClass     DeviceUUID                                      DeviceName
 210    iPhone          xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx        iPhone von Stefan

**Comment by stoff99** [comment]
@llorenzo , if you copy the files 
/lib/aot/bin/iOSBin/idb
and
/lib/aot/bin/iOSBin/iIPA
from build version 86 or 98 to 176 its working again.

But i'm not sure if there was changes between in this 2 files ...

**Comment by ajwfrost** [comment]
@stoff99  thanks - so this is another issue that's caused by us signing the executable files .. but it looks like this can't then load in the (third party) library files! Frustrating - apologies! Yes please grab the earlier files. We'll likely push out an updated SDK that doesn't have these signed binaries, so that most folk can continue, and the Catalina users will need to manually approve the unsigned binaries for running..

FYI in that folder, the only change should have been an updated version of ideviceinstaller which now supports iPhone XR.

**Comment by stoff99** [comment]
ok great, thx for your fast answer.

Just a litte side question: Is your new launch image storyboard overwriting my launch image storyboard if i have the license not active right now?

Thx Stefan

**Comment by ajwfrost** [comment]
Yes unfortunately we couldn't find a way to sort out launch screens that meant free-tier developers can also include their own launch storyboards.

**Comment by ukmunnithan** [comment]
Hello ,
I am also getting the same issue with Flash builder 4.7. The iOS device is not detecting. For Earlier version I got the same issue and it works well for 33.1.1.98. I got the same issue again for this release too..

**Comment by ukmunnithan** [comment]
Hello @ajwfrost @marchbold @stoff99 

I have copied /lib/aot/bin/iOSBin/idb
and
/lib/aot/bin/iOSBin/iIPA
from build version 98 to 176 ,Now the device is detected But when running the application we are getting the following error


Error occurred while packaging the application:


2020-07-06 12:04:31.588 java[3016:55096] -[NSPersistentUIWindowSnapshotter writeWindowSnapshot:length:width:height:bytesPerRow:toFile:inDirectory:encryptingWithKey:uuid:checksum:fd:]: 0 == ftruncate(fd, finalFileSize) failed on line 797: No such file or directory
2020-07-06 12:04:31.588 java[3016:55096] -[NSPersistentUIWindowSnapshotter writeWindowSnapshot:length:width:height:bytesPerRow:toFile:inDirectory:encryptingWithKey:uuid:checksum:fd:]: 0 == ftruncate(fd, fileLength) failed on line 868: No such file or directory

**Comment by Devkjk** [comment]
I do above.
And detected device but publishing last timing "invalid error" dialog show.

MacOS10.14.6
Animate2020
AIRSDK33.1.1.176
Debug at iphone7 iOS13

**Comment by ajwfrost** [comment]
@Devkjk I'm not sure what you mean, are you able to add a screenshot of the dialog please?

thanks

**Comment by Devkjk** [comment]
Correctly 「入力が無効です。」
<img width="534" alt="スクリーンショット 2020-07-09 23 27 26" src="https://user-images.githubusercontent.com/67408099/87052675-feaf9400-c23b-11ea-88c9-d00a798897a8.png">

**Comment by ajwfrost** [comment]
Hmm.. 'the input is invalid' makes me think there's something else going on there. Are you able to launch a terminal window and navigate to your SDK's "bin" folder, and then run:
``` ./adt -devices -platform ios```

thanks

**Comment by Devkjk** [comment]
My iPhone device info appears.

**Comment by Devkjk** [comment]
<img width="626" alt="スクリーンショット 2020-07-09 23 42 02" src="https://user-images.githubusercontent.com/67408099/87054469-1ee05280-c23e-11ea-8358-20cb284d97ed.png">

**Comment by Devkjk** [comment]
When I change to AIRSDK33.1.1.98, publish is ok.

**Comment by ajwfrost** [comment]
Okay so that part looks okay - presumably you get the same response whether you use .98 or .176 for this list of devices?

And if you have an IPA file somewhere, can you please use the command line to see if you can install it:
```./adt -installApp -platform ios -device 9 package xxx.ipa```

thanks

**Comment by Devkjk** [comment]
.98 or .176 detected device is same.
The command result is "Invalid argument: package"

**Comment by Devkjk** [comment]
oh not package but -package.

**Comment by Devkjk** [comment]
Install is OK.

**Comment by ajwfrost** [comment]
@Devkjk  Just coming back to this one: so you can install using 33.1.1.176 and the command line (sorry, yes it should have been "-package") but then when using Animate, it's still giving you the "invalid input" error? I'm wondering if this is internal to Animate as I don't think that's how they would report an error from ADT; can you please try adding a line to the "adt.cfg" file that should be in your SDK's 'lib' folder:
`DebugOut=true`
And then after using Animate, you may have a file in your home directory called 'adt.log' that gives us some more clues..

**Comment by Devkjk** [comment]
Hi,ajwfrost.
I sent the file by email that address is "temporary~gmail".

**Comment by ajwfrost** [comment]
@Devkjk  thanks yes I just got that .. and I should have thought of this, as this was the other big problem we had in the .176 build:
`Invalid input: Compilation failed while executing : compile-abc`

The compile-abc program wasn't signed with the right entitlements.. we're about to release an update that will fix this. Alternatively it should work if you copy the compile-abc and compile-abc-64 files from the .98 release..

Thanks

**Comment by Devkjk** [comment]
@ajwfrost if copy the compile-abc and compile-abc-64 files from the .98, other error occured  when publishing.
<img width="488" alt="スクリーンショット 2020-07-12 21 37 16" src="https://user-images.githubusercontent.com/67408099/87246419-5cd0b700-c488-11ea-9d79-355c9197b397.png">
